### PR TITLE
Add test for nx graph and gar

### PR DESCRIPTION
**Pull request by lnfjpt**
NX graph:
  1. vertex only
  2. vertex & edges
  3. vertex & edges with property
  4. vertex & edges with large property
  5. vertex & edges with multi-label and property
  6. error case
gar:
  1. invalid yaml
  2. unexpected structure
  3. exception
loader:
  1. csv file for pandas

### CODE REVIEW

File: gar/sssp/.gar_config.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+components:
```
Comment (yecol) [inline gar/sssp/.gar_config.yaml:1]: thanks neng, however, the format is outdated. maybe leave the gar related to dongze.

File: test_loader/g_pandas.csv [hunk 2]
```diff
@@ -0,0 +1,17 @@
+1,2,8
```
Comment (yecol) [inline test_loader/g_pandas.csv:1]: also add a g_pandas_named.csv

File: test_nx/nx_g.e1 [hunk 3]
```diff
@@ -0,0 +1,2 @@
+src,dst
+1,3
```
Comment (yecol) [inline test_nx/nx_g.e1:2]: what's this for?

File: test_nx/nx_g.e2_error_parallel_edge [hunk 4]
```diff
@@ -0,0 +1,4 @@
+src,dst
```
Comment (yecol) [inline test_nx/nx_g.e2_error_parallel_edge:1]: what's the name `e2_error_parallel_edge` indicating?

File: test_nx/nx_g_large_property.e [hunk 5]
```diff
@@ -0,0 +1,18 @@
+src,dst,p1,p2,p3,p4,p5,p6,p7,p8,p9,p10,p11,p12,p13,p14,p15
```
Comment (yecol) [inline test_nx/nx_g_large_property.e:1]: format is incorrect....

Comment (yecol) [reply test_nx/nx_g_large_property.e:1]: a property graph has many files, each file record ONE type of vertices/edges with the same label.
```
# a file named person.v for vertices with label person
vid username gender age ...
1 Jim male 24
2 XX male 20
```

```
# a file named company.v for vertices with label company
vid name location founded_time
1 Tencent SZ 1999
2 Ali HZ 1999
```

```
# a file named works_in.py for edges with label works_in
srcid dstid hire_time
1 1 1999
2 1 2020
```
... so a graph with X-label edges and Y-label vertices should have X+Y files.

Comment (lnfjpt) [reply test_nx/nx_g_large_property.e:1]: nx_g_large_property.e and nx_g_large_property.v are used for nx graph test :
```
        gs_g = load_from(vertices={
            'v': ([("vid", VId.INT), ('p1', 'string'), ('p2', 'int'),
                   ('p3', 'string'), ('p4', 'double'), ('p5', 'double'),
                   ('p6', 'int'), ('p7', 'string'), ('p8', 'string'),
                   ('p9', 'string'), ('p10', 'double'), ('p11', 'string'),
                   ('p12', 'float'), ('p13', 'string'), ('p14', 'int'),
                   ('p15', 'double')], loader(vfile_large_property))
        },
                       edges={
                           'e': ('v', 'v', 'both_in_out', [("src", SrcId.INT),
                                                           ("dst", DstId.INT),
                                                           ('p1', 'string'),
                                                           ('p2', 'int'),
                                                           ('p3', 'string'),
                                                           ('p4', 'double'),
                                                           ('p5', 'double'),
                                                           ('p6', 'int'),
                                                           ('p7', 'string'),
                                                           ('p8', 'string'),
                                                           ('p9', 'string'),
                                                           ('p10', 'double'),
                                                           ('p11', 'string'),
                                                           ('p12', 'float'),
                                                           ('p13', 'string'),
                                                           ('p14', 'int'),
                                                           ('p15', 'double')],
                                 loader(efile_large_property))
                       })
```
p1-15 in edge file are different from those in vertex file.
I will make this part more clear.
Should I simplify this part and make these properties more clear?

File: test_nx/nx_g.e1 (hunk 3 above)
Comment (lnfjpt) [reply test_nx/nx_g.e1:2]: This file is used for NX graph test, 
```
g4 = load_from(vertices={}, edges={}
            'v1': ([("id", VId.INT)], loader(vfile1)),
            'v2': ([("id", VId.INT)], loader(vfile2)),
            'v3': ([("id", VId.INT)], loader(vfile3)),
            'v4': ([("id", VId.INT)], loader(vfile4))
        },
        edges={
            'e1': ('v1', 'v2', 'both_out_in', [("src", SrcId.INT), ("dst", DstId.INT)], loader(efile)),
            'e2':('v3', 'v4', 'both_out_in', [("src", SrcId.INT),("dst", DstId.INT)],loader(efile))})
```
Therefore, there are four vertex files and two edge files.

File: gar/sssp/.gar_config.yaml (hunk 1 above)
Comment (lnfjpt) [reply gar/sssp/.gar_config.yaml:1]: I'll remove this part .

File: test_nx/nx_g.e2_error_parallel_edge (hunk 4 above)
Comment (lnfjpt) [reply test_nx/nx_g.e2_error_parallel_edge:1]: gs graph should be loaded without parallel edges.
```
        # graph have parallel edge in two edge label
        # load a graph contain parallel edges in 'e1' and  'e2'
        gs_g = load_from(vertices={
            'v1': ([("id", VId.INT)], loader(vfile1)),
            'v2': ([("id", VId.INT)], loader(vfile2)),
            'v3': ([("id", VId.INT)], loader(vfile3)),
            'v4': ([("id", VId.INT)], loader(vfile4))
        },
                       edges={
                           'e1': ('v1', 'v2', 'both_out_in', [
                               ("src", SrcId.INT), ("dst", DstId.INT)
                           ], loader(efile)),
                           'e2':
                           ('v3', 'v4', 'both_out_in', [("src", SrcId.INT),
                                                        ("dst", DstId.INT)],
                            loader(efile))
                       })
        with pytest.raises(RuntimeError):
            nx_g = self.Graph(gs_g)
```
It seems that this part should be improved. I'll check this part with weibin

File: test_nx/nx_g.e1 (hunk 3 above)
Comment (yecol) [reply test_nx/nx_g.e1:2]: ignore this case, this case can use a property-graph.

**Pull request closed without merging**